www/nghttp2/Portfile

```diff
@@ -52,49 +45,86 @@ configure.args      --disable-silent-rules \
                     --enable-app \
                     ac_cv_prog_AWK=/usr/bin/awk
 
-configure.env       JANSSON_CFLAGS=-I${prefix}/include \
-                    LIBEVENT_OPENSSL_CFLAGS=-I${prefix}/include/event2 \
-                    OPENSSL_CFLAGS=-I${prefix}/include/openssl
+subport libnghttp2 {
+    configure.args-replace \
+                    --enable-app --enable-lib-only
 
-configure.env-append \
-                    "JANSSON_LIBS=-L${prefix}/lib -ljansson" \
-                    "LIBEVENT_OPENSSL_LIBS=-L${prefix}/lib -levent -levent_openssl" \
-                    "OPENSSL_LIBS=-L${prefix}/lib -lcrypto -lssl"
+    description     nghttp2 is an implementation of HTTP/2 in C\; library only.
+    long_description ${description}
 
-if {[variant_isset python27]} { set PythonVersion   2.7 }
-if {[variant_isset python36]} { set PythonVersion   3.6 }
-if {[variant_isset python37]} { set PythonVersion   3.7 }
-if {[variant_isset python38]} { set PythonVersion   3.8 }
+    build.dir       ${worksrcpath}/lib
+    destroot.dir    ${worksrcpath}/lib
 
-if {[variant_isset python27] || [variant_isset python36] || [variant_isset python37] || [variant_isset python38]} {
-    set PythonBranch    [join [lrange [split ${PythonVersion} .] 0 1] ""]
+    pre-activate {
+        # address upgrading from an earlier version which contained
+        # the libraries
+        if {[active_variants file:lib/libnghttp2.dylib:nghttp2 {}]} {
```

Comment:
I'm not familiar with this method. It may not be correct, since the builds are failing:
```
DEBUG: Checking nghttp2 for active variants for depspec 'file:lib/libnghttp2.dylib:nghttp2'
Error: Failed to activate libnghttp2: Registry error: nghttp2 not registered as installed & active.
```
As I mentioned before, the boilerplate we usually use in this situation is at https://trac.macports.org/wiki/PortfileRecipes#deactivatehack

Reply:
Yeah, I fixed that afterwards; the thing is, we should _only_ deactivate the port when it actually conflicts, and I think I managed that. While at it, I moved over all dependencies and added another change that factored the Python bindings into subports.

Reply:
> we should _only_ deactivate the port when it actually conflicts

When would it not conflict?

Reply:
The next upgrade, basically; I'm moving a library from one port to a new dependency. A thought, though: the tools are somewhat esoteric, so it might be better to just factor them out. Especially if all the dependent ports build with this change.

Reply:
> The next upgrade, basically; I'm moving a library from one port to a new dependency

Exactly. That is what the PortfileRecipes entry I referred you to is designed to help you solve. Specifically, this is the code I would suggest using:

```
pre-activate {
    # Files now installed by libnghttp2 used to be installed by nghttp2.
    # This deactivate hack can be removed after September 2021.
    if {![catch {set installed [lindex [registry_active nghttp2] 0]}]} {
        set installed_version [lindex ${installed} 1]
        set installed_revision [lindex ${installed} 2]
        if {[vercmp ${installed_version} 1.41.0] < 0 || ([vercmp ${installed_version} 1.41.0] == 0 && ${installed_revision} < 1)} {
            registry_deactivate_composite nghttp2 "" [list ports_nodepcheck 1]
        }
    }
}
```

Reply:
The current code is this, though:

```tcl
         # address upgrading from an earlier version which contained
         # the libraries in the main port -- added September 2020
         if {[registry_file_registered ${prefix}/lib/libnghttp2.a] eq nghttp2} {
             registry_deactivate_composite nghttp2 "" [list ports_nodepcheck 1]
         }
```

Personally, I find it simpler and a bit more explicit, but there's a value to consistency, of course. I can change it, if you prefer?

Reply:
Your code does look simpler. I do not know if it has the same effect as the method that Clemens documented in the wiki in 2013 and that we have been using since then. @neverpanic, what do you think?

Reply:
> Your code does look simpler. I do not know if it has the same effect as the method that Clemens documented in the wiki in 2013 and that we have been using since then. @neverpanic, what do you think?

Sometimes, too much discussion indicates that you're doing it wrong, and this felt like one of those times, so I changed it to separate the tools rather than the library into a subport. Since all dependencies built with just the library _anyway_, there wasn't much point in keeping the old name for the tools.

Perhaps the code above might be useful for something else later on. Anyway, is it okay to push this once the tests pass?

Reply:
I think both approaches are fine – I used the version number back then, but if you have a specific file you can check for, that's OK, too. I like the simplicity of your approach.